examples/main/main.cpp

```diff
@@ -182,6 +186,10 @@ int main(int argc, char ** argv) {
     g_model = &model;
     g_ctx = &ctx;
 
+#if defined(GGML_USE_KOMPUTE)
+    ggml_vk_init_device(0, "gpu");
+#endif
```

Comment:
This should be moved in `llama_backend_init()`

Reply:
This should not be required to use the backend, it should be done automatically.

Reply:
It's done in test-backend-ops via ggml_backend_reg_kompute_init. Without this, the backend will fail to initialize.

I suppose this should be changed to accept a device index in ggml_backend_kompute_init() like how CUDA does.

Reply:
Adding multi-GPU support to the backend would be the best solution, but until then, this should be done transparently in the entry point functions, possibly when creating a backend and when allocating a buffer.

Reply:
In my opinion it is not ok to require application developers to do this. We cannot have a backend that is only usable in GPT4All.

Reply:
I'm not saying that we shouldn't remove the Kompute-specific code in 'main' - my point is that we can't just initialize the first GPU device automatically without breaking GPT4All: https://github.com/nomic-ai/gpt4all/blob/0a45dd384e29de4e8e282935a206e874b6ddd569/gpt4all-chat/chatllm.cpp#L291

Reply:
You can do the initialization automatically on the default device only if it was not already initialized by a call to `ggml_vk_init_device`.